light-client/src/builder/light_client.rs

```diff
@@ -0,0 +1,213 @@
+            peer_id: self.peer_id,
+            options: self.options,
+            io: self.io,
+            clock: self.clock,
+            hasher: self.hasher,
+            verifier: self.verifier,
+            scheduler: self.scheduler,
+            predicates: self.predicates,
+            light_store: self.light_store,
+            state,
+        }
+    }
+}
+
+impl LightClientBuilder<NoTrustedState> {
+    /// Initialize a builder for a production (non-mock) light client.
+    #[cfg(feature = "rpc-client")]
+    pub fn prod(
+        peer_id: PeerId,
+        rpc_client: rpc::HttpClient,
+        light_store: Box<dyn LightStore>,
+        options: Options,
+        timeout: Option<Duration>,
+    ) -> Self {
+        Self::custom(
+            peer_id,
+            options,
+            light_store,
+            Box::new(ProdIo::new(peer_id, rpc_client, timeout)),
+            Box::new(ProdHasher),
+            Box::new(SystemClock),
+            Box::new(ProdVerifier::default()),
+            Box::new(scheduler::basic_bisecting_schedule),
+            Box::new(ProdPredicates),
+        )
+    }
+
+    /// Initialize a builder for a custom light client, by providing all dependencies upfront.
+    #[allow(clippy::too_many_arguments)]
+    pub fn custom(
+        peer_id: PeerId,
+        options: Options,
+        light_store: Box<dyn LightStore>,
+        io: Box<dyn Io>,
+        hasher: Box<dyn Hasher>,
+        clock: Box<dyn Clock>,
+        verifier: Box<dyn Verifier>,
+        scheduler: Box<dyn Scheduler>,
+        predicates: Box<dyn VerificationPredicates>,
+    ) -> Self {
+        Self {
+            peer_id,
+            hasher,
+            io,
+            verifier,
+            light_store,
+            clock,
+            scheduler,
+            options,
+            predicates,
+            state: NoTrustedState,
+        }
+    }
+
+    /// Set the given light block as the initial trusted state.
+    pub fn trust_light_block(
+        mut self,
+        trusted_state: LightBlock,
+    ) -> Result<LightClientBuilder<HasTrustedState>, Error> {
+        self.validate(&trusted_state)?;
+
+        // TODO(liamsi, romac): it is unclear if this should be Trusted or only Verified
+        self.light_store.insert(trusted_state, Status::Trusted);
+
+        Ok(self.with_state(HasTrustedState))
+    }
+
+    /// Keep using the latest verified or trusted block in the light store.
+    /// Such a block must exists otherwise this will fail.
+    pub fn trust_from_store(self) -> Result<LightClientBuilder<HasTrustedState>, Error> {
+        let trusted_state = self
+            .light_store
+            .latest_trusted_or_verified()
+            .ok_or_else(|| error::Kind::NoTrustedStateInStore)?;
+
+        self.trust_light_block(trusted_state)
+    }
+
+    /// Fetch and set the latest block from the primary peer as the trusted state.
+    pub fn trust_primary_latest(self) -> Result<LightClientBuilder<HasTrustedState>, Error> {
```

Comment:
This is a bit dangerous and should either come with a big warning or be removed all together

Reply:
Let's remove it since we don't have any use for it at the moment.

Reply:
0e1709d